packages/leancode_lint/lib/lints/avoid_missing_dispose.dart

```diff
@@ -0,0 +1,365 @@
+// Replace this once analyzer is updated
+// ignore_for_file: deprecated_member_use
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element2.dart';
+import 'package:analyzer/dart/element/type.dart';
+import 'package:analyzer/error/error.dart' hide LintCode;
+import 'package:analyzer/error/listener.dart';
+import 'package:custom_lint_builder/custom_lint_builder.dart';
+import 'package:leancode_lint/utils.dart';
+import 'package:yaml/yaml.dart';
+
+class _IgnoredInstance {
+  const _IgnoredInstance({required this.name, required this.packageName});
+
+  final String name;
+  final String packageName;
+}
+
+class AvoidMissingDisposeConfig {
+  const AvoidMissingDisposeConfig({
+    this.ignoredInstances = const {},
+    this.ignoredInstancesCheckers = const [],
+  });
+
+  factory AvoidMissingDisposeConfig.fromConfig(Map<String, Object?> json) {
+    final ignoredInstances =
+        (json['ignored_instances'] as YamlList?)?.nodes
+            .map(
+              (e) => _IgnoredInstance(
+                name: (e as YamlMap)['ignore'] as String,
+                packageName: e['from_package'] as String,
+              ),
+            )
+            .toSet() ??
+        const {};
+    return AvoidMissingDisposeConfig(
+      ignoredInstances: ignoredInstances,
+      ignoredInstancesCheckers: [
+        for (final _IgnoredInstance(:name, :packageName) in ignoredInstances)
+          if (packageName.startsWith('dart:'))
+            TypeChecker.fromUrl('$packageName#$name')
+          else
+            TypeChecker.fromName(name, packageName: packageName),
+      ],
+    );
+  }
+
+  final Set<_IgnoredInstance> ignoredInstances;
```

Comment:
```suggestion
  final Set<_IgnoredTypes> ignoredTypes;
```
Rename `instances` to `types` across this PR